### DISCUSSION TITLE
How to set tooltips for code blocks?

### DISCUSSION THREAD

**Discussion by ssbarnea** [squidfunk/mkdocs-material#5204] [Q&A] [answered]
While we can configure `title=...` for code blocks, we currently cannot add HTML native tooltips via `tooltip=...`, which can prove very useful, as tooltips do not take screen space.

Any chance to get this added?

**Comment by squidfunk** [comment, marked as answer]
Thanks for suggesting. I'm not sure what you mean. Where should the tooltip appear? Below the entire code block? If you could provide a mock up or more detailed explanation, that would be very helpful for evaluating if this might be a fit for the project.

**Reply by ssbarnea** [reply]
HTML browser are able to display tooltip attribute by themselves on almost any visible element. You don't need to write any code for tooltip, only to add the *title* attribute to the top element (div?) of the block will make it work.

See https://www.w3schools.com/tags/att_global_title.asp -- that is basically what tooltip does. Extra bonus is that it also improves the accessibility part.

**Reply by squidfunk** [reply]
If you add a `title` attribute to any element inside the Markdown part of a document, the tooltip logic will display it. Unfortunately, `title` is already used by Pygments to display a title above the code block, so that won't work, but this will:

```` markdown
<div title="Custom tooltip" markdown>

``` yaml
markdown_extensions:
  - pymdownx.highlight:
      anchor_linenums: true
      line_spans: __span
      pygments_lang_class: true
  - pymdownx.inlinehilite
  - pymdownx.snippets
  - pymdownx.superfences
```

</div>
````

Result:

<img width="724" alt="Bildschirmfoto 2023-03-16 um 21 03 20" src="https://user-images.githubusercontent.com/932156/225740101-c4cab28d-73e4-4127-8bb9-08cf44d0dfe1.png">

**Reply by squidfunk** [reply]
I'm marking this as an answer, since it solves the problem at hand.
